### DISCUSSION TITLE
Retrieve localisation for "dashboard" string in toolbar from friends controller array

### DISCUSSION THREAD

**Discussion by Loreos7** [ppy/osu#20711] [Q&A] [answered]
Currently its:
![image](https://user-images.githubusercontent.com/86934170/195071713-e31f9a9c-84a5-441b-969c-f06a1bbe3efd.png)
But should be:
![image](https://user-images.githubusercontent.com/86934170/195071770-012d81c5-011b-4903-bfd2-069fc903bd6a.png)
Because of:
![image](https://user-images.githubusercontent.com/86934170/195071911-25d36a3b-9a7f-4cf1-935d-a10e0deb1265.png)
![image](https://user-images.githubusercontent.com/86934170/195072136-23311f74-78e6-4aaf-b5c4-7369f15a7960.png)
Due to conflicts in localizations.

**Comment by peppy** [comment]
This seems a bit weird. Can you share a localisation where these two don't match for context?

**Reply by Loreos7** [reply]
![image](https://user-images.githubusercontent.com/86934170/195408283-f1de6370-1eb3-4d12-9f44-2070b1a61757.png)
![image](https://user-images.githubusercontent.com/86934170/195408482-41b3f5eb-c8e4-460f-a743-d1c8d128a932.png)

I think I've pinned wrong screenshot for the current. I guess the actual is:
![image](https://user-images.githubusercontent.com/86934170/195408774-5e02e328-44a8-47b5-82c8-4cc398ba1dda.png)

**Reply by Loreos7** [reply]
For Ukranian: 
![image](https://user-images.githubusercontent.com/86934170/195409874-02380749-79ad-4f92-be6a-e2b051ef13cd.png)
but
![image](https://user-images.githubusercontent.com/86934170/195409907-70ded362-be42-4521-9a20-139f8f574365.png)

**Comment by frenzibyte** [comment, marked as answer]
Apparently osu!web generates a localisable string for every controller that leads to the same page, so there are about 5 "dashboard" localisable strings each assigned to the controller that is being dealt with.
```
resources/lang/en/page_title.php-        'topic_watches_controller' => [
resources/lang/en/page_title.php:            'index' => 'dashboard',
resources/lang/en/page_title.php-        ],
--
resources/lang/en/page_title.php-        'beatmapset_watches_controller' => [
resources/lang/en/page_title.php:            'index' => 'dashboard',
resources/lang/en/page_title.php-        ],
--
resources/lang/en/page_title.php-        'follows_controller' => [
resources/lang/en/page_title.php:            'index' => 'dashboard',
resources/lang/en/page_title.php-        ],
resources/lang/en/page_title.php-        'friends_controller' => [
resources/lang/en/page_title.php:            'index' => 'dashboard',
resources/lang/en/page_title.php-        ],
--
resources/lang/en/page_title.php-        'home_controller' => [
resources/lang/en/page_title.php-            'get_download' => 'download',
resources/lang/en/page_title.php:            'index' => 'dashboard',
resources/lang/en/page_title.php-            'search' => 'search',
resources/lang/en/page_title.php-            'support_the_game' => 'support the game',
```
<img width="1095" alt="CleanShot 2022-10-12 at 16 18 21@2x" src="https://user-images.githubusercontent.com/22781491/195353206-c2126b10-0fe5-4d86-85c3-eb43cd3cb0d8.png">

That's quite awkward to deal with in lazer, given that an overlay's title is supposed to remain constant regardless of selected tab. I feel it's an oversight that more than one localisable string exists in the first place, but can't say for sure.

**Reply by peppy** [reply]
cc/ @nanaya @notbakaneko , can this potentially be fixed osu-web side? not high priority but a weird one.

**Reply by nanaya** [reply]
it'll be fixed by ppy/osu-web#9392 but please also check the usage here and make sure to use the ones that's being [kept around](https://github.com/ppy/osu-web/pull/9392/files#diff-e7252f2afe8038968ede9adb0cc88dd0337201cadb82d2e1d376d0638b7aeb5eR832-R850)

**Reply by Loreos7** [reply]
> it'll be fixed by [ppy/osu-web#9392](https://github.com/ppy/osu-web/pull/9392) but please also check the usage here and make sure to use the ones that's being [kept around](https://github.com/ppy/osu-web/pull/9392/files#diff-e7252f2afe8038968ede9adb0cc88dd0337201cadb82d2e1d376d0638b7aeb5eR832-R850)

I don't understand why you removed every "dashboard" varieties, even from the web site. you should have just prefered `main.friends_controller.index` for game button.

**Reply by frenzibyte** [reply]
They're unnecessary duplicates.
